ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/PluginBuilder.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("DEPRECATION")
+
+package io.ktor.server.application
+
+import io.ktor.server.application.debug.*
+import io.ktor.server.config.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import io.ktor.util.debug.*
+import io.ktor.util.pipeline.*
+import kotlinx.coroutines.*
+import kotlin.random.*
+
+/**
+ * A builder that is available inside a plugin creation block. Using this builder you can hook in a different steps of request processing.
+ **/
+@PluginsDslMarker
+public interface PluginBuilder<PluginConfig : Any> {
+
+    /**
+     * Reference to the [Application] where plugin is installed.
+     */
+    public val application: Application
+
+    /**
+     * Specifies handler [block] for every incoming [ApplicationCall].
+     *
+     * This block will be invoked for every incoming call even if the call was handled by some different handler.
```

Comment:
will be invoked -> is invoked
was handled -> is already handled

Reply:
Fixed